src/libs/TransactionUtils/index.ts

```diff
@@ -827,6 +828,24 @@ function getRateID(transaction: OnyxInputOrEntry<Transaction>): string | undefin
     return transaction?.modifiedCustomUnitRateID ?? transaction?.comment?.customUnit?.customUnitRateID?.toString();
 }
 
+/**
+ * Whether expense requests connected with given transaction IDs are unsubmitted by the owners
+ */
+function areRequestsInTransactionsUnsubmitted(transactionIDs: string[]): boolean {
+    const allReports = ReportConnection.getAllReports();
```

Comment:
We can't rely on Search data being stored in Onyx, in this case `report_` key. Search can return results that you currently don't have in Onyx and we might not update those results with push updates for you.

Reply:
Could you please give me an example of a case where the search data for the expense report isn't stored in Onyx? I'd reproduce it and research whether there's any other way of handling this.

Reply:
I don't have clear reproducible steps, but I think the most common case is Old reports that are not fetched in OpenApp/ReconnectApp.

Reply:
I've discussed with @Kicu about the search flow and the most reliable data for this. We are currently using transaction data to determine if we can show the hold option.

For example, in this screenshot:

<img width="868" alt="Screenshot 2024-08-27 at 10 17 46" src="https://github.com/user-attachments/assets/e76761f4-3616-47fe-99eb-eca6548f2946">
We use the canHold parameter stored in the transaction objects. To address the issue raised, we’d need to update this parameter on both the frontend and backend to set it to false when the transaction is unsubmitted.

Given my understanding of the search system and data reliability in the search flow, this seems like the only approach which will work even for the cases you've listed. What do you think?

Reply:
That seems correct to me. 

>To address the issue raised, we’d need to update this parameter on both the frontend and backend to set it to false when the transaction is unsubmitted.

We still allow transactions to be held/unheld on unsubmitted reports if you're the owner of the transaction. Admins can only hold/unhold on submitted transactions. I think this is already reflected in the backend code.

Reply:
We currently have this logic in the backend, so we allow unsubmitted transactions to be held if it's on an IOU report or if the user is the owner of the expense, otherwise only submitted transactions can be held

```
const setInt& allowedStates = (isIOU || isOwner) ? setInt{Report::STATE_OPEN, Report::STATE_SUBMITTED} : setInt{Report::STATE_SUBMITTED};
```

Reply:
So, to conclude- it sounds like we would respect the `canHold` parameter in its existing form, and would not deliberately set it to `false` if the transaction is unsubmitted (for the purposes of Search), is that correct?

Reply:
Yea. Is the `canHold` value returned by the API incorrect?

Reply:
If we want to completely hide the hold option if one of the selected transactions is unsubmitted ([as discussed here](https://github.com/Expensify/App/issues/46742#issuecomment-2290075501)), then yes, the value would be incorrect.

Outside of the search flow, we use the `ReportUtils.canHoldUnholdReportAction(reportAction: OnyxInputOrEntry<ReportAction>)` util function to determine whether the report action is holdable, and it returns the `{canHold: boolean, canUnhold: boolean}` values back to us. This I assume uses the same logic as the backend for setting them for transactions.

However, we also have additional frontend logic to check whether the report action is unsubmitted and will throw an additional error on the `HoldReasonPage.tsx`, and this part is where the logic between the backend for the transaction value and the frontend for hold differs. It was added because we didn't want to disable the hold option altogether, but we still want to block the user from holding and throw an error message.

<img width="1086" alt="Screenshot 2024-08-28 at 10 54 19" src="https://github.com/user-attachments/assets/f0dc7785-7c92-4a70-b1d7-1617759eb530">

So if we want to include this case here, as was requested by the issue I'm trying to fix with this PR, we'd need to update the backend to include it as well.

Reply:
I'm not sure that I follow this. Maybe we can take a step back. These are the steps to reproduce the issue:

1. [Employee] Go to workspace chat
2. [Employee] Create two expenses and do not click Submit button
3. [Admin] Go to transaction thread of any submitted expense from employee
4. [Admin] Click on the report header > Hold
5. [Admin] Enter reason and save it
6. [Admin] Note that it shows "This request is being modified by another member." and the expense cannot be held
7. [Admin] Go to Search
8. [Admin] Select the expense from Step 4 via checkbox
9. [Admin] Click on the dropdown > Hold
10. [Admin] Enter reason and save it
11. [Admin] Note that admin can hold the request from Search and there is no error message

From these steps, it sounds like the only issue here is that the backend is returning `canHold: true` when it should return `canHold: false`. That being said, we can see from the logic I shared above that we only allow admins to hold expenses on processing (submitted) reports. Can we confirm the state and status of the report when we try to hold it from the report header?

Reply:
cc @robertjchen @carlosmiceli 

Hm maybe this has been already patched on the backend side? I tested it now and it looks like we get the correct canHold value, which is being set to false for the unsubmitted reports. It has not worked like this when I was creating this PR / when the issue was raised so since then there may have been changes to correct it.

https://github.com/user-attachments/assets/2039fa3b-3546-4134-b0e0-cedb4b9c7848

If we have changed it then I think we can close this PR and the issue itself as it has been resolved on the backend.

Reply:
cc @luacmartins

Reply:
Sorry, wrong cc 😅

Reply:
We did update the `canHold` logic and deployed it on Aug 15. Maybe this was fixed as part of that.

Reply:
I've created fresh accounts and recreated the issue steps on main, I've attached a video of it. Looks like currently this has been fixed on the backend side, can confirm that we can close the issue.

https://github.com/user-attachments/assets/1d4c2ae0-28c5-444e-94f3-4a5ab5e9137c

cc @luacmartins

Reply:
Thanks @cdOut. Sorry for the extra work here.